kyo-sttp/jvm/src/main/scala/kyo/interop/Adapter.scala

```diff
@@ -0,0 +1,401 @@
+package kyo.interop
+
+import java.util.concurrent.Flow.*
+import kyo.*
+import kyo.Emit.Ack
+import kyo.Maybe.*
+import kyo.Result.*
+import kyo.kernel.ArrowEffect
+import scala.annotation.tailrec
+
+object Adapters:
+
+    private trait KyoSubscriber[A] extends Subscriber[A]:
+        def isDone: Boolean < IO
+        def await: Unit < (Async & Abort[SubscriberDone])
+        def interrupt: Unit < IO
+    end KyoSubscriber
+
+    abstract class SubscriberDone
+    case object SubscriberDone extends SubscriberDone
+
+    private def makeSubscriber[A](
+        bufferSize: Int
+    )(
+        using Frame
+    ): (KyoSubscriber[A], Fiber.Promise[Nothing, (Subscription, Queue[A])]) < IO =
+        IO.Unsafe.apply {
+            val queue              = Queue.Unsafe.init[A](capacity = bufferSize, access = Access.MultiProducerMultiConsumer)
+            val onSubscribePromise = Fiber.Promise.Unsafe.init[Nothing, (Subscription, Queue[A])]()
+            val kyoSubscriber = new KyoSubscriber[A]:
```

Comment:
The mix of safe and unsafe code can be tricky as the codebase evolves. It'd be nice to refactor this to a separate class without the `AllowUnsafe` evidence from `IO.Unsafe` in scope. Otherwise there's a higher risk of accidentally  omitting IO suspensions.